src/components/ProductDetail.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import Carousel from "./Carousel";
+const _ = require("lodash");
+
+
+class ProductDetail extends React.Component {
+  render() {
+    const paper = this.props;
+    const ProductList = paper.props.products.map(function (product, index) {
+			// cannot figure out how to give these keys without it throwing errors in the code.
+      const initialRatings = [
+        <span key={index + product.id + "a"} className="glyphicon glyphicon-star" />,
+        <span key={index + product.id + "b"} className="glyphicon glyphicon-star" />,
+        <span key={index + product.id + "c"} className="glyphicon glyphicon-star" />,
+        <span key={index + product.id + "d"} className="glyphicon glyphicon-star" />,
+        <span key={index + product.id + "e"} className="glyphicon glyphicon-star" />
```

Comment:
While this works, you usually would do something like that with an array. Lodash has a function to generate an array. For example:  `_.range(5).map(convertToStar)`.